src/core/include/iDynTree/Core/Rotation.h

```diff
@@ -164,7 +164,7 @@ namespace iDynTree
          /**
          * Get a roll, pitch and yaw corresponding to this rotation.
          *
-         * Get \f$ r \in [-π, π] , p \in [-π/2, π/2], y \in [-π, π]\f$
+         * Get \f$ (r,p,y) \in (-π, π] \times (-π/2, π/2) \times (-π, π] \bigcap  \{0\} \times \{-π/2,π/2\} \times (-π,π)\f$
```

Comment:
Modifications:
* Are you use that you want to use the "intersection" of sets? I think we should have a union of sets there, so `\bigcap` --> `\bigcup` 
* As all the symbols are regular size, I think it is more readable if also the set union is regular size, so `\bigcup` --> `\cup`. 
* Add parenthesis to be explicit on the precedence between the cartesian product operator and the set union operator (I guess most users are not aware about the consensus about precedence between these operators)
* In the configuration set corresponding to singularity, I guess the yaw can also take the exact value `\pi`, as it is computed using the `atan2` function, right? So I added `]` in the last `(-\pi, \pi)`.
See the following suggestion:
```suggestion
         * Get \f$ (r,p,y) \in ((-π, π] \times (-π/2, π/2) \times (-π, π]) \cup ( \{0\} \times \{-π/2,π/2\} \times (-π,π] )\f$
```

Reply:
Yes sorry, it is the union. Also according to the source code:
```
if (R(2, 0) < 1.0) {
        if (R(2, 0) > -1.0) {
            r = atan2(R(2, 1), R(2, 2));
            p = asin(-R(2, 0));
            y = atan2(R(1, 0), R(0, 0));
        } else {
            // Not a unique solution
            // Singularity
            r = 0.0;
            p = IDJL_M_PI / 2.0;
            y = -atan2(-R(1, 2), R(1, 1));
        }
    } else {
        // Not a unique solution
        // Singularity
        r = 0.0;
        p = -IDJL_M_PI / 2.0;
        y = atan2(-R(1, 2), R(1, 1));
    }
```
We should have two separate cases handling the singularity.

Reply:
> We should have two separate cases handling the singularity.

Indeed. That may be nice, as the `\{-π/2,π/2\}` part was not super understandable in any case.